workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/generate/HyscaleGenerateServiceManifestsCommand.java

```diff
@@ -90,64 +92,80 @@
-    
+    private ServiceSpecProcessor serviceSpecProcessor;
+
     @Autowired
-    private PostValidatorsProvider postValidatorsProvider;
-    
+    private RegistryValidator registryValidator;
+
     @Autowired
-    private WorkflowContextBuilder workflowContextBuilder;
-    
+    private ManifestValidator manifestValidator;
+
+    private List<Validator<WorkflowContext>> postValidators;
+
+    @PostConstruct
+    public void init() {
+        this.postValidators = new LinkedList<>();
+        this.postValidators.add(registryValidator);
+        this.postValidators.add(manifestValidator);
+    }
+
     @Override
     public Integer call() throws Exception {
         WorkflowLogger.header(ControllerActivity.PROCESSING_INPUT);
         if (!CommandUtil.isInputValid(this)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
-        
-        HyscaleCommandSpecBuilder commandSpecBuilder = new HyscaleCommandSpecBuilder();
-        commandSpecBuilder.setAppName(appName);
-        commandSpecBuilder.setServiceSpecFiles(serviceSpecs);
-        if (profileArg != null) {
-            commandSpecBuilder.setProfileFiles(profileArg.getProfiles());
-            commandSpecBuilder.setProfileName(profileArg.getProfileName());
-        }
 
-        // Handles input preprocessing
-        HyscaleInputSpec hyscaleInput = hyscaleInputSpecProcessor.process(commandSpecBuilder);
-        if (hyscaleInput == null) {
+        // Validate Service specs with schema
+        if (!serviceSpecInputValidator.validate(serviceSpecsFiles)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
-        
+
         Map<String, File> serviceVsSpecFile = new HashMap<String, File>();
-        for (File serviceSpec : serviceSpecs) {
+        for (File serviceSpec : serviceSpecsFiles) {
             serviceVsSpecFile.put(ServiceSpecUtil.getServiceName(serviceSpec), serviceSpec);
         }
-        List<EffectiveServiceSpec> effectiveServiceSpecList = serviceSpecProcessor.getEffectiveServiceSpec(hyscaleInput);
-        
-        List<WorkflowContext> contextList = workflowContextBuilder.buildContextList(effectiveServiceSpecList, appName, null);
-        
-        postValidatorsProvider.getManifestPostValidators().forEach( each -> inputSpecPostValidator.addValidator(each));
-        
+
+        // Construct EffectiveServiceSpec from ProfileArg & ServiceSpecFiles
+        List<EffectiveServiceSpec> effectiveServiceSpecs = serviceSpecProcessor.process(profileArg, serviceSpecsFiles);
+
+        List<WorkflowContext> contextList = effectiveServiceSpecs.stream().filter(each -> {
+            return each != null && each.getServiceSpec() != null && each.getServiceMetadata() != null;
+        }).map(each -> {
+            WorkflowContextBuilder builder = new WorkflowContextBuilder(appName);
+            try {
+                builder.withProfile(each.getServiceMetadata().getEnvName());
+                builder.withService(each.getServiceSpec());
```

Comment:
Should we have a separate method for service spec, and we can set the service name separately using effective service spec. This would avoid fetching data from service spec again.

Reply:
That is already there, this is just to make sure that even if user forgets to set service name even in case of service spec, it's an automated setting through servicespec